File: Niveau 4/3 – Structures de données et Balayages/1 - Parc d'attraction.cpp
#include <iostream>
#include <vector>

using namespace std;

int main()
{
    ios::sync_with_stdio(false);
    int nbJour, nbPeriode;

    vector<int> periodes = {0};

    cin >> nbJour >> nbPeriode;
    int somme = 0;
    for (int i = 0; i < nbJour; i++)
    {
        int visiteur;
        cin >> visiteur;
        somme += visiteur;
        periodes.push_back(somme);
    }

    for (int i = 0; i < nbPeriode; i++)
    {
        int start, end;
        cin >> start >> end;
        cout << periodes[end] - periodes[start - 1] << '\n';
    }
}

File: Niveau 4/3 – Structures de données et Balayages/1 - Parc d'attraction.py
import sys
def main():
   nbJour, nbPeriode = map(int, sys.stdin.readline().split())
   periodes = [0]*(nbJour + 1)
   somme = 0
   for j, n in enumerate(sys.stdin.readline().split(), 1):
      somme += int(n)
      periodes[j] = somme
   
   for i in range(nbPeriode):
      start, end = map(int, sys.stdin.readline().split())
      sys.stdout.write(str(periodes[end] - periodes[start - 1]) + "\n")
main()